tests/instances/eth_runner/src/single_run.rs

```diff
@@ -0,0 +1,132 @@
+use crate::block::Block;
+use crate::block_hashes::BlockHashes;
+use crate::post_check::post_check;
+use crate::prestate::{populate_prestate, DiffTrace, PrestateTrace};
+use crate::receipts::{BlockReceipts, TransactionReceipt};
+use rig::log::info;
+use rig::*;
+use std::fs::{self, File};
+use std::io::BufReader;
+
+#[allow(clippy::too_many_arguments)]
+fn run<const RANDOMIZED: bool>(
+    mut chain: Chain<RANDOMIZED>,
+    block_context: BlockContext,
+    block_number: u64,
+    miner: alloy::primitives::Address,
+    ps_trace: PrestateTrace,
+    transactions: Vec<Vec<u8>>,
+    receipts: Vec<TransactionReceipt>,
+    diff_trace: DiffTrace,
+    block_hashes: Option<BlockHashes>,
+) -> anyhow::Result<()> {
+    chain.set_last_block_number(block_number - 1);
+
+    if let Some(block_hashes) = block_hashes {
+        chain.set_block_hashes(block_hashes.into_array(block_number))
+    }
+
+    let prestate_cache = populate_prestate(&mut chain, ps_trace);
+
+    let output = chain.run_block(transactions, Some(block_context), None);
+
+    post_check(
+        output,
+        receipts,
+        diff_trace,
+        prestate_cache,
+        ruint::aliases::B160::from_be_bytes(miner.into()),
+    )
+    .unwrap();
+
+    Ok(())
+}
+
+pub fn single_run(
+    block: String,
+    calltrace: String,
+    receipts: String,
+    prestatetrace: String,
+    difftrace: String,
+    block_hashes: Option<String>,
+    randomized: bool,
+) -> anyhow::Result<()> {
+    let block = fs::read_to_string(block)?;
+    // TODO: ensure there are no calls to unsupported precompiles
```

Comment:
Btw we also have blobbasefee and blobhash

Reply:
That's gonna be a bit more annoying, as we need opcode tracing,no?

Reply:
But _generally_, we could run and trace blocks using Revm